packages/database/src/changeHandlers/EntityHierarchyCacher.js

```diff
@@ -3,55 +3,46 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 import { reduceToDictionary } from '@tupaia/utils';
+import winston from 'winston';
 import { ORG_UNIT_ENTITY_TYPES } from '../modelClasses/Entity';
+import { ChangeHandler } from './ChangeHandler';
 
-const REBUILD_DEBOUNCE_TIME = 1000; // wait 1 second after changes before rebuilding, to avoid double-up
-
-export class EntityHierarchyCacher {
+export class EntityHierarchyCacher extends ChangeHandler {
```

Comment:
I hadn't realised that `EntityHierarchyCacher` wasn't a `ChangeHandler` when I took on this work 😅 

It was a bit of a tricky refactor, but the unit tests really saved my life. It's good to have consistency, and I feel things are a bit simpler now that it doesn't do any of the promise management stuff itself!

Reply:
So much simpler, I love it! Thanks for doing this. Also really like that the change handlers are no longer async, that was clever.

One thing that feels a little like clutter is passing `transactingModels` so many places. I'm not sure if it would make things more or less clear, but did you think of trying a separate class that handles processing a rebuild, and gets constructed at that time with the `transactingModels` so each function can still refer to `this.transactingModels`?

Reply:
Yeah that's a good call, I'll break it out and see how it looks 👍